worker/src/unstract/worker/worker_interface.py

```diff
@@ -0,0 +1,77 @@
+from abc import ABC, abstractmethod
+
+
+class UnstratcWorkerInterface(ABC):
+    @abstractmethod
+    def __init__(self, image_name: str, image_tag: str):
+        pass
+
+    @abstractmethod
+    def _get_image(self) -> str:
+        pass
+
+    @abstractmethod
+    def _image_exists(self, image_name_with_tag: str) -> bool:
+        pass
+
+    @abstractmethod
+    def normalize_container_name(self, name: str) -> str:
+        pass
+
+    @abstractmethod
+    def stream_logs(self, container):
+        pass
+
+    @abstractmethod
+    def get_valid_log_message(self, log_line: str):
+        pass
+
+    @abstractmethod
+    def process_log_message(self, log_line: str, channel):
+        pass
+
+    @abstractmethod
+    def is_valid_log_type(self, log_type: str) -> bool:
+        pass
+
+    @abstractmethod
+    def get_log_type(self, log_dict: dict) -> str:
+        pass
```

Comment:
All these 4 methods should be common to both k8s and docker. This is our own logics. So please add implementation here instead of making this an abstract method.